src/ui/popup.js

```diff
@@ -107,11 +107,13 @@ export default class Popup extends Evented {
     _lngLat: LngLat;
     _trackPointer: boolean;
     _pos: ?Point;
+    _classList: Set<string>;
 
     constructor(options: PopupOptions) {
         super();
         this.options = extend(Object.create(defaultOptions), options);
         bindAll(['_update', '_onClose', 'remove', '_onMouseMove', '_onMouseUp', '_onDrag'], this);
+        this._classList = new Set(options && options.className ? options.className.split(` `) : []);
```

Comment:
Maybe trim and split by one _or more_ whitespace to increase robustness (e.g. silly input like `"  foo   bar "`).

```suggestion
        this._classList = new Set(options && options.className ? options.className.trim().split(/\s+/) : []);
```

Reply:
Good call!